components/dropdown/basic/src/index.js

```diff
@@ -134,13 +134,17 @@ class DropdownBasic extends Component {
     const wrapperClassName = cx('sui-DropdownBasic', {
       'is-expanded': expanded
     })
+    const buttonWrapperClassName = cx('sui-DropdownBasic-buttonWrap', {
+      [button.className]: !!button.className
```

Comment:
Classname brake the encapsulation of this component.

You must use an enum prop and apply the inner class properly depend on the enum value.

Reply:
Just for info... why do you need a custom classname?

Reply:
He needs a custom className to just use a `re-Select` class in order to give to the dropdown the look'n'feel of Fotocasa. This is not ideal, as the re-Select itself should be a component instead just classnames to be used.

Reply:
As you say, it's not the ideal, and as far as I know, we've been taken a different approach for this specific cases by overriding the classnames from our sites containers... So, I don't know if it's a door we should open, aren't you happy by overriding classnames?